File: src/core/interfaces/types/types/navigation.ts
import { ScreenType } from '../utils/screenNavigation';

export interface GameState {
  currentLocation?: {
    hasPOI: boolean;
    poiId?: string;
  };
  party?: {
    members: string[];
  };
  inventory?: {
    hasItems: boolean;
  };
}

export interface NavigationState {
  currentScreen: ScreenType;
  previousScreen?: ScreenType;
  history: ScreenType[];
  isTransitioning: boolean;
}

export interface NavigationOptions {
  animate?: boolean;
  skipAccessibilityCheck?: boolean;
  preserveHistory?: boolean;
}

export type NavigationDirection = 'next' | 'previous' | 'specific';

File: backend/app/src/core/interfaces/types/types/navigation.py
from typing import Any, List, Union


class GameState:
    currentLocation?: {
    hasPOI: bool
    poiId?: str
  party?: {
    members: List[string]
  }
  inventory?: {
    hasItems: bool
  }
}
class NavigationState:
    currentScreen: ScreenType
    previousScreen?: ScreenType
    history: List[ScreenType]
    isTransitioning: bool
class NavigationOptions:
    animate?: bool
    skipAccessibilityCheck?: bool
    preserveHistory?: bool
NavigationDirection = Union['next', 'previous', 'specific']